lib/slack/real_time/config.rb

```diff
@@ -15,6 +18,23 @@ def reset
         self.websocket_ping = 30
         self.websocket_proxy = nil
         self.token = nil
+        self.concurrency = method(:detect_concurrency)
+      end
+
+      def concurrency
+        (val = @concurrency).respond_to?(:call) ? val.call : val
```

Comment:
I did not do it for speed, but for being dry. I really dislike repeating long instance variable names because they just return nil if you mistype them.